app/controllers/api/v1/facts_controller.rb

```diff
@@ -0,0 +1,76 @@
+class Api::V1::FactsController < ApplicationController
+    include AuthenticationCheck
+  
+    before_action :is_user_logged_in
+    before_action :set_fact, only: [:show, :update, :destroy]
+    before_action :check_access
+  
+    # GET /members/:member_id/facts
+    def index
+      @member = Member.find(params[:member_id])
+      render json: @member.facts # note that because the facts route is nested inside members
+                               # we return only the facts belonging to that member
+    end
+  
+    # GET /members/:member_id/facts/:id
+    def show
+        if @fact
+          render json: @fact
+        else
+          render json: { error: 'Fact not found' }, status: 404
```

Comment:
Because a similar error is repeated in multiple actions, I would consider moving the `if @fact` check to the `set_fact` method.